Documentation on tensor parallelism (#339)
huggingface/optimum-neuron: docs/source/package_reference/distributed.mdx

<!--Copyright 2023 The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
the License. You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
-->
# Optimum Neuron Distributed

The `optimum.neuron.distributed` module provides a set of tools to perform distributed training and inference.

## Parallelization

> The main task in distributed training / inference is being able to shard things such as the model weights, the gradient and the optimizer state. The `Parallelizer` classes handle that.

Reviewer: ```suggestion
The main task in distributed training / inference is to shard the model weights, the gradient, or/and the optimizer state. We build `Parallelizer` classes to handle the sharding.
```
